src/crewai/cli/tools/main.py

```diff
@@ -103,23 +103,40 @@ def install(self, handle: str):
             )
             raise SystemExit
 
-        self._add_repository_to_poetry(get_response.json())
         self._add_package(get_response.json())
 
         console.print(f"Succesfully installed {handle}", style="bold green")
 
-    def _add_repository_to_poetry(self, tool_details):
-        repository_handle = f"crewai-{tool_details['repository']['handle']}"
-        repository_url = tool_details["repository"]["url"]
-        repository_credentials = tool_details["repository"]["credentials"]
+    def login(self):
+        login_response = self.plus_api_client.login_to_tool_repository()
+
+        if login_response.status_code != 200:
+            console.print(
+                "Failed to authenticate to the tool repository. Make sure you have the access to tools.",
+                style="bold red",
+            )
+            raise SystemExit
```

Comment:
Would be better to handle auth and general errors separately. Better feedback reduces the chance of users asking for support. Is that possible here?

Reply:
I see! Do you mean using a custom exception?

Reply:
I'm not sure if I understand, but I like the idea of improving feedback. Could you please elaborate?